src/Microsoft.OData.Core/SelectedPropertiesNode.cs

```diff
@@ -469,6 +503,44 @@ private static IEnumerable<string> GetPossibleMatchesForSelectedOperation(IEdmOp
             yield return qualifiedContainerName + operationNameWithParameters;
         }
 
+        private bool IsValidExpandToken(string item)
+        {
+            // Check #1: Expand token must contain '(', must be related to navigation property.
+            int idxRP = item.IndexOf('(');
+            if (idxRP == -1
+                || (this.structuredType != null &&
+                    !(this.structuredType.FindProperty(item.Substring(0, idxRP)) is IEdmNavigationProperty)

```

Comment:
>!(this.structuredType.FindProperty(item.Substring(0, idxRP)) [](start = 20, length = 60)

As discussed, this does not allow for "dynamic" navigation properties.  It's probably sufficient to say that, if it contains an open paren, then it represents an expand (and must terminate in a balanced closing paren.)

Reply:
The check for "is IEdmNavigationProperty" was put in to guard against Action()/Function() alike (see comment in L517 below), which has parens. Looks like we need to further resolve potential ambiguity between navProp() and Action()/Function() alike.

Reply:
Per our discussion offline today, in order to avoid introducing other ambiguities related to Function/Action resolution, we don't support dynamic navigation property for now.